subsys/net/ip/net_context.c

```diff
@@ -341,6 +359,12 @@ int net_context_put(struct net_context *context)
 		return 0;
 	}
 
+#if defined(CONFIG_NET_TLS) || defined(CONFIG_NET_DTLS)
+	if (context->tls) {
```

Comment:
What will finalize TLS structures for normally-closed TCP connection (one for which net_tcp_put() returns 0)?

This looks like a serious mistake (worth -1 on its own).

Reply:
I agree, it doesn't look good and should be fixed.